profile/profile.go

```diff
@@ -134,6 +154,19 @@ func initEventMap(profile *profctx.Profile) {
 			common.TRIGGER_AN_RELEASE_EVENT:   common.CONNECTION_RELEASE_REQUEST_EVENT,
 			common.PROFILE_PASS_EVENT:         common.QUIT_EVENT,
 		}
+	case NW_TRIGG_UE_DEREG_REQ:
+		profile.Events = map[common.EventType]common.EventType{
+			common.REG_REQUEST_EVENT:           common.AUTH_REQUEST_EVENT,
+			common.AUTH_REQUEST_EVENT:          common.AUTH_RESPONSE_EVENT,
+			common.SEC_MOD_COMMAND_EVENT:       common.SEC_MOD_COMPLETE_EVENT,
+			common.REG_ACCEPT_EVENT:            common.REG_COMPLETE_EVENT,
+			common.PDU_SESS_EST_REQUEST_EVENT:  common.PDU_SESS_EST_ACCEPT_EVENT,
+			common.PDU_SESS_EST_ACCEPT_EVENT:   common.PDU_SESS_EST_ACCEPT_EVENT,
+			common.DEREG_REQUEST_UE_TERM_EVENT: common.DEREG_ACCEPT_UE_TERM_EVENT,
+			common.DEREG_ACCEPT_UE_TERM_EVENT:  common.DEREG_ACCEPT_UE_TERM_EVENT,
```

Comment:
We did a similar thing for PDU_SESS_EST_ACCEPT_EVENT as it is an incoming NAS message. So that we check with the event map what is the required action for it.

I believe DEREG_ACCEPT_UE_TERM is an outgoing message and thus we do not need it to be used as KEY in the event map. Line 165 is the right change. Please correct me if I am wrong